cmd/create/operatorroles/cmd.go

```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
 			os.Exit(1)
 		}
 	}
+	policies, err := r.OCMClient.GetPolicies("OperatorRole")
+	if err != nil {
+		r.Reporter.Errorf("Expected a valid role creation mode: %s", err)
+		os.Exit(1)
+	}
 
-	roleName, err := aws.GetInstallerAccountRoleName(cluster)
+	defaultPolicyVersion, err := r.OCMClient.GetDefaultVersion()
 	if err != nil {
-		r.Reporter.Errorf("Expected parsing role account role '%s': %v", cluster.AWS().STS().RoleARN(), err)
+		r.Reporter.Errorf("Error getting latest default version: %s", err)
 		os.Exit(1)
 	}
 
-	path, err := getPathFromInstallerRole(cluster)
+	if args.prefix != "" {
+		return handleOperatorRoleCreationByPrefix(r, env, permissionsBoundary,
+			mode, policies, defaultPolicyVersion)
+	}
+	return handleOperatorRoleCreationByClusterKey(r, env, permissionsBoundary,
+		mode, policies, defaultPolicyVersion)
+}
+
+func handleOperatorRoleCreationByPrefix(r *rosa.Runtime, env string,
+	permissionsBoundary string, mode string,
+	policies map[string]*cmv1.AWSSTSPolicy,
+	defaultPolicyVersion string) error {
+	includeHostedCpSet := args.hostedCp
+	operatorRolesPrefix := args.prefix
+	if len(operatorRolesPrefix) == 0 {
+		r.Reporter.Errorf("Expected a prefix for the operator IAM roles")
+		os.Exit(1)
+	}
+	if len(operatorRolesPrefix) > 32 {
+		r.Reporter.Errorf("Expected a prefix with no more than 32 characters")
+		os.Exit(1)
+	}
+	if !aws.RoleNameRE.MatchString(operatorRolesPrefix) {
+		r.Reporter.Errorf("Expected valid operator roles prefix matching %s", aws.RoleNameRE.String())
+		os.Exit(1)
+	}
+	oidcEndpointUrl := args.oidcEndpointUrl
+	if oidcEndpointUrl == "" {
+		r.Reporter.Errorf("%s is mandatory for %s param flow.", OidcEndpointUrlFlag, PrefixFlag)
+		os.Exit(1)
+	}
+	parsedURI, err := url.ParseRequestURI(oidcEndpointUrl)
 	if err != nil {
-		r.Reporter.Errorf("Expected a valid path for '%s': %v", cluster.AWS().STS().RoleARN(), err)
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	if parsedURI.Scheme != helper.ProtocolHttps {
+		r.Reporter.Errorf("Expected OIDC endpoint URL '%s' to use an https:// scheme", oidcEndpointUrl)
+		os.Exit(1)
+	}
+	installerRoleArn := args.installerRoleArn
+	if installerRoleArn == "" {
```

Comment:
Same here, we can check the flags at the beginning of the command.
If the user provided `PrefixFlag`, we can check they provided the other required flags for BYO OIDC.